nevergrad/optimization/optimizerlib.py

```diff
@@ -677,22 +685,22 @@ def __init__(
             parametrization: IntOrParameter,
             budget: Optional[int] = None,
             num_workers: int = 1,
-            num_optims: Optional[int] = None,
-            num_vars: Optional[List[Any]] = None,
-            multivariate_optimizer: base.OptimizerFamily = CMA,
-            monovariate_optimizer: base.OptimizerFamily = RandomSearch
+            num_optims: tp.Optional[int] = None,
+            num_vars: Optional[List[int]] = None,
+            multivariate_optimizer: ConfigOptim = CMA,
+            monovariate_optimizer: ConfigOptim = RandomSearch
     ) -> None:
         super().__init__(parametrization, budget=budget, num_workers=num_workers)
-        if num_vars:
-            if num_optims:
+        if num_vars is not None:
```

Comment:
@teytaud make `if num_vars` can lead to unexpected behavior if what you wanted is to test agains `None`, so always use `if num_vars is not None`